sdk/eventhubs/azure-messaging-eventhubs/src/producer_client.cpp

```diff
@@ -99,7 +99,7 @@ namespace Azure { namespace Messaging { namespace EventHubs {
     auto batch = CreateBatch(EventDataBatchOptions{}, context);
     for (const auto& data : eventData)
     {
-      if (!batch.TryAddMessage(data))
+      if (!batch.TryAdd(data))
```

Comment:
i like the TryAddMessage better than just TryAdd

Reply:
But in this case you're not adding a message, you're adding a Batch.

FWIW, these are the names from the various libraries:

Language|Method Name | Notes
-----|----|---
Python | EventDataBatch.add| Throws if it fails.
Java|EventDataBatch.tryAdd| Only accepts an EventData
javascript|EventDataBatch.tryAdd| Accepts an EventData or AnnotatedAmqpMessage
go|EventDataBatch.AddEventData, EventDataBatch.AddAMQPAnnotatedMessage| Go does not support method overloading.
.net| EventDataBatch.TryAdd | Only accepts an EventData

I went with `TryAdd` because it aligned with the majority of our existing language SDKs which support method overloads.